File: ruby/examples/post_shopping_cart.rb
require_relative '../request.rb'

# Example of POST /shopping-cart

COMPANY_ID			= ENV['COMPANY_ID']
FILING_PRODUCT_ID 	= ENV['FILING_PRODUCT_ID']
FILING_METHOD_ID 	= ENV['FILING_METHOD_ID']
QUANTITY = 1
# const FORM_DATA = {}; # optional, expects JSON mapping to fields of filing method schema

body = {
	company_id: COMPANY_ID,
	product_id: FILING_PRODUCT_ID,
	product_option_id: FILING_METHOD_ID,
	quantity: QUANTITY,
	#form_data: FORM_DATA
}.to_json

BaseRequestRoute.request(:post, '/shopping-cart', body: body)

File: javascript/examples/post_shopping_cart.js
const baseRequest = require('../base_request.js').baseRequest

const path = require('path');
process.chdir(path.resolve(__dirname, '../../'));
require('dotenv').config();

// Example of POST /shopping-cart

const COMPANY_ID		= process.env.COMPANY_ID;
const FILING_PRODUCT_ID	= process.env.FILING_PRODUCT_ID;
const FILING_METHOD_ID 	= process.env.FILING_METHOD_ID;
const QUANTITY = 1;
// const FORM_DATA = {}; // optional, expects JSON mapping to fields of filing method schema

let body = { 
	company_id: COMPANY_ID,
	product_id: FILING_PRODUCT_ID,
	product_option_id: FILING_METHOD_ID,
	quantity: QUANTITY,
	// form_data: FORM_DATA
};

token = baseRequest.request.token({ path: '/shopping-cart', body: body });

baseRequest.request.post({ path: '/shopping-cart', token: token, body: body });